Update "What's New" and "Glossary" (#404)
amzn/ion-docs: _books/ion-1-1/src/glossary.md

See _encoding expression_.

**encoding directive**<br/>

> In an Ion 1.1 segment, a top-level E-expression annotated with `$ion`.

Reviewer: ```suggestion
In an Ion 1.1 segment, a top-level E-expression that invokes the implicit `$ion` macro.
```
Or however we want to phrase the effect of `(:$ion <directive_name> ...)`.

A 1-byte, unsigned integer that tells the reader what the next expression represents
and how the bytes that follow should be interpreted.

**parameter**<br/>
An input to a macro, as defined by its signature.
At expansion time a parameter produces zero or one values.

**qualified macro reference**<br/>
A macro reference that consists of a module name and either a macro name exported by that module,
or a numeric address within the range of the module's exported macro table. In text, these look
like :_module-name_::_name-or-address_.

**segment**<br/>
A contiguous partition of a _document_ that uses the same _encoding module sequence_.
Segment boundaries are caused by directives: an IVM, `set_symbols`, `add_symbols`, `set_macros`, `add_macros`, `use`, and `encoding` directives end segments (with a new one starting immediately afterward).
The `import` and `module` directives can also end a segment if they are redefining a module binding that was in the encoding module sequence.

**shared module**<br/>
A module that exists independent of the data stream of an Ion document. It is identified by a
name and version so that it can be imported by other modules.

**signature**<br/>

> The (derived) part of a macro definition that specifies its "calling convention", in terms of the shape and type of arguments it accepts.

Reviewer: Should we define 'placeholder' and then define signature in terms of placeholders? Or just drop 'signature'...

Author: Yes, I'll do that.
